docs/api-docs/provider-apis/tax/overview.mdx

```diff
@@ -143,6 +143,16 @@ BigCommerce sends requests for tax estimates on product item, shipping, and hand
 
 [View the Estimate Taxes API reference](/docs/rest-contracts/tax#estimate-taxes).
 
+### Performance
+
+Merchants expect fast tax estimate responses to facilitate a streamlined checkout experience for their shoppers, hence performance is an essential consideration for a tax provider.
+
+To protect the shopper checkout experience from overly slow tax responses, tax estimate requests will timeout after a set period of time. The default threshold of four seconds is intended to support a wide variety of use cases (e.g. large tax estimate requests).
```

Comment:
I've highlighted the _large cart_ scenario here since I think it's important to consider that not every tax request received will be equal - it may vary be merchant.

Indeed, this might be a legitimate reason to split out a tax provider's service into multiple components: one that services small carts, one that services large... and having appropriate timeouts set for each. That is, one size may not fit all.

That said, I think going into too much detail here could be excessive, and so I'm keen to keep it relatively light.

Reply:
We've allowed variations of 4-10 seconds in cases where we've highlighted exactly the above and the tax provider has opted to proceed with their ask. The most extreme case was the Midwest Goods tax provider (IGEN, from memory) that wanted to go even beyond 10 seconds, which we refused. That is, in their case reliable tax estimates (on large carts) were more important than speed.

So far these have been limited availability providers only, available only on a handful of stores - and so that seems acceptable to me, as long as the merchant knows what they are getting into and understands the consequences.

We've considered whether it's worth highlighting these timeout values - or performance metrics more generally - in the UI to give merchants better visibility into what they are getting into... with the intent of removing BigCommerce from the support conversation and directly encouraging tax providers to lift their performance. That is, taking more of an 'enablement' stance rather than getting in the way.